src/lib.rs

```diff
@@ -145,6 +161,26 @@ impl Sdl2Window {
         Ok(window)
     }
 
+    /// Initialize the joystick subsystem. Required before joystick input
+    /// events will be returned. Returns the number available or error.
+    pub fn init_joysticks(&mut self) -> Result<u32, String> {
```

Comment:
Each time a new event is emitted, the number of devices might have changed. I wonder if applications should just listen to events, instead of detecting devices you configure the application.